packages/drivers/odsp-driver/src/test/socketTests/socketTests.spec.ts

```diff
@@ -490,4 +490,61 @@ describe("OdspDocumentDeltaConnection tests", () => {
 			"1 get_ops_response listener should exiist",
 		);
 	});
+
+	it("Multiple connection objects should handle server_disconnect event when the second client is still waiting for connection to complete", async () => {
+		socket = new ClientSocketMock();
+		const connection1 = await mockSocket(socket as unknown as Socket, async () =>
+			OdspDocumentDeltaConnection.create(
+				tenantId,
+				documentId,
+				token,
+				client,
+				webSocketUrl,
+				logger,
+				60000,
+				epochTracker,
+				socketReferenceKeyPrefix,
+			),
+		);
+
+		socket.setMockSocketConnectResponseForReuse({
+			connect_document: { eventToEmit: undefined },
+		});
+		let connection2Fails = false;
+		let errorReceived: IAnyDriverError | undefined;
+		const connection2 = mockSocket(socket as unknown as Socket, async () =>
+			OdspDocumentDeltaConnection.create(
+				tenantId,
+				documentId,
+				token,
+				client,
+				webSocketUrl,
+				logger,
+				60000,
+				epochTracker,
+				socketReferenceKeyPrefix,
+			),
+		).catch((error: IAnyDriverError) => {
+			connection2Fails = true;
+			errorReceived = error;
+		});
+		let disconnectedEvent1 = false;
+
+		const errorToThrow = { message: "OdspSocketError", code: 400 };
+		connection1.on("disconnect", (reason: IAnyDriverError) => {
+			disconnectedEvent1 = true;
+		});
+
+		socket.sendServerDisconnectEvent(errorToThrow);
+
+		assert(disconnectedEvent1, "disconnect event should happen on first object");
+
+		assert(socket !== undefined && !socket.connected, "socket should be disconnected");
+		checkListenerCount(socket);
+		await connection2;
+		assert(connection2Fails, "connection2 should fail");
+		assert(errorReceived?.message.includes("server_disconnect"), "message should be correct");
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+		assert((errorReceived as any).statusCode === 400, "status code should be correct");
```

Comment:
Let's avoid `any` in new code, I think the right thing to cast as here would be a `IGenericNetworkError`?